boards/arm/ip_k66f/doc/index.rst

```diff
@@ -0,0 +1,175 @@
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| USB       | on-chip    | USB device                          |
++-----------+------------+-------------------------------------+
+| WATCHDOG  | on-chip    | watchdog                            |
++-----------+------------+-------------------------------------+
+| FLASH     | on-chip    | soc flash                           |
++-----------+------------+-------------------------------------+
+| RTC       | on-chip    | rtc                                 |
++-----------+------------+-------------------------------------+
+
+The default configuration can be found in the defconfig file:
+
+	``boards/arm/ip_k66f/ip_k66f_defconfig``
+
+Micrel/Microchip KSZ8794CNX Ethernet Switch is not currently
+supported.
+
+Connections and IOs
+===================
+
+The K66F SoC has five pairs of pinmux/gpio controllers.
+
++-------+-----------------+---------------------------+
+| Name  | Function        | Usage                     |
++=======+=================+===========================+
+| PTA8  | GPIO            | Red LED                   |
++-------+-----------------+---------------------------+
+| PTA10 | GPIO            | RED LED                   |
++-------+-----------------+---------------------------+
+
+System Clock
+============
+
+The K66F SoC is configured to use the 12 MHz low gain crystal oscillator on the
+board with the on-chip PLL to generate a 180 MHz system clock.
+
+Serial Port
+===========
+
+The K66F SoC has six UARTs. None is used.
+
+USB
+===
+
+The K66F SoC has a USB OTG (USBOTG) controller that supports both
+device and host functions through its micro USB connector (K66F USB).
+Only USB device function is supported in Zephyr at the moment.
+
+Programming and Debugging
+*************************
+
+Build and flash applications as usual (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+Configuring a Debug Probe
+=========================
+
+A debug probe is used for both flashing and debugging the board. This board is
+configured by default to use the :ref:`opensda-jlink-onboard-debug-probe`.
+
+Option 1: :ref:`opensda-jlink-onboard-debug-probe`
+--------------------------------------------------
+
+Install the :ref:`jlink-debug-host-tools` and make sure they are in your search
+path.
+
+Follow the instructions in :ref:`opensda-jlink-onboard-debug-probe` to program
+the `OpenSDA J-Link Generic Firmware for V3.2 Bootloader`_. Note that Segger
+does provide an OpenSDA J-Link Board-Specific Firmware for this board, however
+it is not compatible with the DAPLink bootloader.
+
+Add the argument ``-DOPENSDA_FW=jlink`` when you invoke ``west build`` to
```

Comment:
Perhaps this option could be added as a default for the board if it is always the same?

BTW, is this the normal way to set the runner, I have used the `west flash -r jlink ....` for this. @mbolivar what is the suggested way the runner should be set?

Reply:
@jukkar Am not sure how to make this default. Adding `set(ZEPHYR_BOARD_FLASH_RUNNER jlink)` to board's Cmake doesn't have any impact.

Reply:
Here is an example from frdm_k64f showing how to use `board_set_flasher_ifnset()` and `board_set_debugger_ifnset()` to set the default runners while still respecting any choice the user makes at CMake time.

```
set_ifndef(OPENSDA_FW daplink)

if(OPENSDA_FW STREQUAL jlink)
  board_set_debugger_ifnset(jlink)
  board_set_flasher_ifnset(jlink)
elseif(OPENSDA_FW STREQUAL daplink)
  board_set_debugger_ifnset(pyocd)
  board_set_flasher_ifnset(pyocd)
endif()

board_runner_args(jlink "--device=MK64FN1M0xxx12")
board_runner_args(pyocd "--target=k64f")

include(${ZEPHYR_BASE}/boards/common/pyocd.board.cmake)
include(${ZEPHYR_BASE}/boards/common/jlink.board.cmake)
include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
```

Note that if you don't use a `board_set_foo_ifnset()` function, then the first `include` line will set the defaults, so for example pyocd.board.cmake would set the default runner to pyocd if not already set.

I personally don't like the continued use of the `OPENSDA_FW` variable as it's vendor specific and there is a more general mechanism available via the runner configuration in CMake, but that is @MaureenHelm 's decision since she maintains these SoCs.

Reply:
@mbolivar Thanks you. Updated the document and board.cmake.